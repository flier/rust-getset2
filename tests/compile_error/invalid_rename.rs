use getset2::Getter;

#[derive(Getter)]
pub struct Struct {
    #[get(rename(?))]
    field: usize,
}

#[derive(Getter)]
pub struct Struct2 {
    #[get(rename("test"))]
    field: usize,
}

#[derive(Getter)]
pub struct Struct3 {
    #[get(rename)]
    field3: usize,
}

fn main() {}
