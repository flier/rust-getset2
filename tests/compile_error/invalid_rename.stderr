error: invalid field name to rename, expected identifier
 --> tests/compile_error/invalid_rename.rs:5:20
  |
5 |     #[get(rename = "?")]
  |                    ^^^

error: invalid #[get(..)] attribute, expected `rename = ...`, found `rename`(...)
  --> tests/compile_error/invalid_rename.rs:11:5
   |
11 |     #[get(rename("test"))]
   |     ^

error: invalid #[get(..)] attribute, expected `rename = ...`, found `rename`
  --> tests/compile_error/invalid_rename.rs:17:5
   |
17 |     #[get(rename)]
   |     ^
